base_layer/core/tests/helpers/sync.rs

```diff
@@ -0,0 +1,261 @@
+        temp_dir.path().to_str().unwrap(),
+    )
+    .await;
+    let shutdown = Shutdown::new();
+    let (state_change_event_publisher, _) = broadcast::channel(10);
+    let (status_event_sender, _status_event_receiver) = watch::channel(StatusInfo::new());
+
+    // Alice needs a state machine for header sync
+    let alice_state_machine = BaseNodeStateMachine::new(
+        alice_node.blockchain_db.clone().into(),
+        alice_node.local_nci.clone(),
+        alice_node.comms.connectivity(),
+        alice_node.comms.peer_manager(),
+        alice_node.chain_metadata_handle.get_event_stream(),
+        BaseNodeStateMachineConfig::default(),
+        SyncValidators::new(MockValidator::new(true), MockValidator::new(true)),
+        status_event_sender,
+        state_change_event_publisher,
+        RandomXFactory::default(),
+        consensus_manager.clone(),
+        shutdown.to_signal(),
+    );
+
+    (
+        alice_state_machine,
+        alice_node,
+        bob_node,
+        initial_block,
+        consensus_manager,
+        key_manager,
+    )
+}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub enum WhatToDelete {
+    Blocks,
+    Headers,
+}
+
+// Delete blocks and headers in reverse order; the first block in the slice wil not be deleted
+pub fn delete_some_blocks_and_headers(
+    blocks_with_anchor: &[ChainBlock],
+    instruction: WhatToDelete,
+    node: &NodeInterfaces,
+    set_best_block: Option<bool>,
+) {
+    if blocks_with_anchor.is_empty() || blocks_with_anchor.len() < 2 {
+        panic!("blocks must have at least 2 elements");
+    }
+    let set_best_block = set_best_block.unwrap_or(false);
+    let mut blocks: Vec<_> = blocks_with_anchor.to_vec();
+    blocks.reverse();
+    for i in 0..blocks.len() - 1 {
+        let mut txn = DbTransaction::new();
+        match instruction {
+            WhatToDelete::Blocks => {
+                txn.delete_block(*blocks[i].hash());
+                txn.delete_orphan(*blocks[i].hash());
+                if set_best_block {
+                    txn.set_best_block(
+                        blocks[i + 1].height(),
+                        blocks[i + 1].accumulated_data().hash,
+                        blocks[i + 1].accumulated_data().total_accumulated_difficulty,
+                        *node.blockchain_db.get_chain_metadata().unwrap().best_block(),
+                        blocks[i + 1].to_chain_header().timestamp(),
+                    );
+                }
+            },
+            WhatToDelete::Headers => {
+                txn.delete_header(blocks[i].height());
+            },
+        }
+        node.blockchain_db.write(txn).unwrap();
+        // Note: Something is funny here... the block is deleted but the block exists in the db
+        // match instruction {
+        //     WhatToDelete::Blocks => {
+        //         assert!(!node.blockchain_db.block_exists(*blocks[i].hash()).unwrap());
+        //     }
+        //     WhatToDelete::Headers => {}
+        // }
```

Comment:
delete?

Reply:
No, I would like us to keep it here, would like to fix it soon, but it is not urgent right away.

Reply:
Can you add a comment exactly saying it there as well, so we know why its commented out and why we left it in